### PR TITLE
[DO NOT MERGE] Upload all docs script

### PR DESCRIPTION
**Pull request by lmossman**
## What
Based on https://github.com/airbytehq/airbyte/pull/30410

Draft PR to add a command to upload the metadata for every connector's current version to GCS.

**Note**: I don't think we should actually merge this PR in, since this shouldn't be needed more than once; instead I think we should keep it as a draft and run the command from this branch, and then close the branch. We can always find this PR again in github if we need the script again for any reason.

## How
Add a command to metadata_service's commands.py which iterates over all connectors and calls the `upload` method for each to upload their metadata to GCS.

For now, I've added a `break` to the loop so it only uploads the metadata for a single connector. Comment this out to upload metadata for all connectors.

But when I actually do run it for real on the prod bucket, I plan to uncomment that last section so that it uploads the docs for all connectors.

Example usage:
```
$ poetry run metadata_service upload-all-metadata "/Users/lakemossman/code/airbyte/" "/Users/lakemossman/code/airbyte/docs/" dev-airbyte-cloud-connector-metadata-service
~~~~~~ Uploading metadata for source-secoda
Running validator: validate_all_tags_are_keyvalue_pairs
Running validator: validate_at_least_one_language_tag
Running validator: validate_major_version_bump_has_breaking_change_entry
Running validator: validate_docs_path_exists
Running validator: validate_metadata_images_in_dockerhub
Checking that the following images are on dockerhub: [('airbyte/source-secoda', '0.1.0')]
Local /Users/lakemossman/code/airbyte/airbyte-integrations/connectors/source-secoda/icon.svg md5_hash: 6QjWZXjaCqp3LDCmCgS3SA==
Remote metadata/airbyte/source-secoda/latest/icon.svg md5_hash: 6QjWZXjaCqp3LDCmCgS3SA==
Local /Users/lakemossman/code/airbyte/airbyte-integrations/connectors/source-secoda/metadata.yaml md5_hash: B8/QzJaHMHoKq6nLVxIBeA==
Remote metadata/airbyte/source-secoda/0.1.0/metadata.yaml md5_hash: B8/QzJaHMHoKq6nLVxIBeA==
Local /Users/lakemossman/code/airbyte/docs/integrations/sources/secoda.md md5_hash: qa3oJHdX3onzbAiWnbVFOA==
Remote metadata/airbyte/source-secoda/0.1.0/doc.md md5_hash: None
Uploading /Users/lakemossman/code/airbyte/docs/integrations/sources/secoda.md to metadata/airbyte/source-secoda/0.1.0/doc.md...
Local /Users/lakemossman/code/airbyte/airbyte-integrations/connectors/source-secoda/metadata.yaml md5_hash: B8/QzJaHMHoKq6nLVxIBeA==
Remote metadata/airbyte/source-secoda/latest/metadata.yaml md5_hash: B8/QzJaHMHoKq6nLVxIBeA==
Local /Users/lakemossman/code/airbyte/docs/integrations/sources/secoda.md md5_hash: qa3oJHdX3onzbAiWnbVFOA==
Remote metadata/airbyte/source-secoda/latest/doc.md md5_hash: None
Uploading /Users/lakemossman/code/airbyte/docs/integrations/sources/secoda.md to metadata/airbyte/source-secoda/latest/doc.md...
```

### CODE REVIEW

**Comment by lmossman** [comment]
@bnchrch could you take a look at the `upload_all_docs` command/script that I've added in this draft PR, and let me know if you have any thoughts? This is what I plan to use to upload all docs for all of the current connector versions after we merge https://github.com/airbytehq/airbyte/pull/30410

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/commands.py
Comment (bnchrch) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/commands.py:69]: ❗Lets be clear this is a dev command by prefixing the command with experimental: `experimental_upload_all_docs`

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py
Comment (bnchrch) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py:266]: 📚 Lets write a docstring explaining what it does, why we added it, and if theres any criteria to remove this

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py
Comment (bnchrch) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py:307]: 💅 would be great to have a single exclusion criteria
💅 would be great not to nest so deep
❗ also I think were missing certain criteria

```py
def is_valid_connector_dir(connector_dir):
  # Check if directory
  # Check starts with source- or destination-
  # Check has metadata.yaml file
  # Do not allow `-scaffold`
  # Do not allow `-secure` or `-strict-encrypt`
  # What about our third party folder?

for connector_dir in connectors_dir.iterdir() if is_valid_connector_dir(connector_dir):
   #...
```

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py
Comment (bnchrch) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py:312]: 💡(Very Optional) if we import ci_connector_ops ([example import](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pyproject.toml#L23)) we can use `Connector` and [get_all_connectors_in_repo](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/connector_ops/connector_ops/utils.py#L475) to do alot of this. Particularly now that the related airbyte-ci PR adds helper properties for getting the doc and inapp doc paths.

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py
Comment (bnchrch) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py:313]: ❓Why did we go with this method instead of just calling `commands.upload(metadata_file_path, doc_path, bucket_name` directly?

**Review by bnchrch — CHANGES_REQUESTED** [review]
A few comments! But the only thing holding this part is making sure we are very explicit and thorough with our connector directory exclusions

**Pull request closed without merging**
